x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/tool/BaseRunAsSuperuserCommand.java

```diff
@@ -0,0 +1,281 @@
+            if (userRoles == null) {
+                throw new IllegalStateException("File realm configuration file [" + rolesFile + "] is missing");
+            }
+            userRoles = new HashMap<>(userRoles);
+            userRoles.put(username, ROLES);
+            FileUserRolesStore.writeFile(userRoles, rolesFile);
+
+            attributesChecker.check(terminal);
+            checkClusterHealthWithRetries(newEnv, terminal, 5);
+            executeCommand(terminal, options, newEnv);
+        } catch (Exception e) {
+            int exitCode;
+            if (e instanceof UserException) {
+                exitCode = ((UserException) e).exitCode;
+            } else {
+                exitCode = ExitCodes.DATA_ERROR;
+            }
+            throw new UserException(exitCode, e.getMessage());
+        } finally {
+            cleanup(terminal, newEnv);
+        }
+    }
+
+    /**
+     * Removes temporary file realm user from users and roles file
+     */
+    protected void cleanup(Terminal terminal, Environment env) throws Exception {
+        final Path passwordFile = FileUserPasswdStore.resolveFile(env);
+        final Path rolesFile = FileUserRolesStore.resolveFile(env);
+        FileAttributesChecker attributesChecker = new FileAttributesChecker(passwordFile, rolesFile);
+
+        Map<String, char[]> users = FileUserPasswdStore.parseFile(passwordFile, null, env.settings());
+        if (users == null) {
+            throw new UserException(ExitCodes.CONFIG, "File realm configuration file [" + passwordFile + "] is missing");
+        }
+        users = new HashMap<>(users);
+        char[] passwd = users.remove(username);
+        if (passwd != null) {
+            // No need to overwrite, if the user was already removed
+            FileUserPasswdStore.writeFile(users, passwordFile);
+            Arrays.fill(passwd, '\0');
+        }
+
+        Map<String, String[]> userRoles = FileUserRolesStore.parseFile(rolesFile, null);
+        if (userRoles == null) {
+            throw new UserException(ExitCodes.CONFIG, "File realm configuration file [" + rolesFile + "] is missing");
+        }
+        userRoles = new HashMap<>(userRoles);
+        String[] roles = userRoles.remove(username);
+        if (roles != null) {
+            // No need to overwrite, if the user was already removed
+            FileUserRolesStore.writeFile(userRoles, rolesFile);
+        }
+        attributesChecker.check(terminal);
+    }
+
+    protected SecureString getPassword() {
+        return password;
+    }
+
+    protected String getUsername() {
+        return username;
+    }
+
+    private void ensureFileRealmEnabled(Settings settings) throws Exception {
+        Map<String, Settings> fileRealmSettings = settings.getGroups("xpack.security.authc.realms.file");
+        if (fileRealmSettings.size() > 1) {
+            throw new UserException(
+                ExitCodes.CONFIG,
+                "Multiple file realms are configured. "
```

Comment:
If possible, lets factor out the code in the server code that tells the file realm configuration is correct and that a file realm is enabled and its order.

Reply:
This check is already happening on server side https://github.com/elastic/elasticsearch/blob/4880ceac90d962d48febf36a998347865bc7a399/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java#L321

Reply:
I can remove the check if you think it's cleaner. The tool needs a running cluster and if a node attempts to start with a configuration file that has more than one file realms, then the node would fail to start because of that, so in theory we would never stumble upon this error in the tool.